solana/programs/ntt-quoter/src/processor/request_relay.rs

```diff
@@ -0,0 +1,138 @@
+use anchor_lang::{
+    prelude::*,
+    system_program::{self, Transfer},
+};
+use example_native_token_transfers::queue::outbox::OutboxItem;
+use solana_program::{native_token::LAMPORTS_PER_SOL, sysvar};
+
+use crate::{
+    error::NttQuoterError,
+    state::{Instance, RegisteredChain, RelayRequest},
+    EVM_GAS_COST, WORMHOLE_TRANSCEIVER_INDEX,
+};
+
+#[derive(Accounts)]
+pub struct RequestRelay<'info> {
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    #[account(seeds = [Instance::SEED_PREFIX], bump = Instance::BUMP)]
+    pub instance: Account<'info, Instance>,
+
+    #[account(mut, address = instance.fee_recipient @ NttQuoterError::InvalidFeeRecipient)]
+    /// CHECK: leave britney alone (anchor complains about a missing check despite the address check)
+    pub fee_recipient: AccountInfo<'info>,
+
+    #[account(
+    seeds = [RegisteredChain::SEED_PREFIX, outbox_item.recipient_chain.id.to_be_bytes().as_ref()],
+    bump = registered_chain.bump,
+    constraint = registered_chain.base_price != u64::MAX @ NttQuoterError::RelayingToChainDisabled
+  )]
+    pub registered_chain: Account<'info, RegisteredChain>,
+
+    //TODO eventually drop the released constraint and instead implement release by relayer
+    #[account(
+    owner = example_native_token_transfers::ID,
+    constraint = outbox_item.released.get(WORMHOLE_TRANSCEIVER_INDEX),
+  )]
+    pub outbox_item: Account<'info, OutboxItem>,
+
+    //TODO should this be init_if_needed?
+    //     are there any security considerations here?
+    //     init_if_needed would allow for multiple requests, even after it was already delivered
+    //     right now requesting a relay for a request is permissionless which would allow
+    //       a third party to initiate a relay without gas dropoff for the user (though they
```

Comment:
if we do `init_if_needed` here, we should make sure that the gas dropoff is strictly higher than the existing gas dropoff request, otherwise the initial gas dropoff payment is lost.
That said, is the relayer going to close this account immediately upon processing the request? If not, then there would be a race condition where the relayer has already processed the request and someone overrides it, which would be ignored by the relayer. I guess that race condition already exists if someone recreates the account (by requesting a new relay) after it's been closed.
I'm assuming the relayer doesn't monitor new relay requests after the initial one (as my assumption is that the relayer just monitors VAAs and looks up their corresponding relay requests based on the transfer ID)

Reply:
> I'm assuming the relayer doesn't monitor new relay requests after the initial one (as my assumption is that the relayer just monitors VAAs and looks up their corresponding relay requests based on the transfer ID)

Correct. My concern was actually primarily around whether someone can prevent an account from being initialized by sending SOL to it (because then in some sense the account is already initialized, even if it has no data), but that was really just an "in the moment confusion" because in that case you could prevent any account that uses `init` and has a predictable address from being created... otoh, I do remember that there was some concern around this in the core bridge rewrite and so I left it in as a reminder to myself while also pointing out that `init_if_needed` would have unexpected consequences too (users being able to pointlessly pay multiple times).